hypershift-operator/controllers/hostedcluster/hostedcluster_controller.go

```diff
@@ -3361,6 +3361,16 @@ func enqueueHostedClustersFunc(metricsSet metrics.MetricsSet, operatorNamespace
 		case *hyperv1.NodePool:
 			return []reconcile.Request{{NamespacedName: types.NamespacedName{Name: typedObj.Spec.ClusterName, Namespace: typedObj.Namespace}}}
 		case *corev1.Secret:
+			// reconcile the karpenter token rotations to the userData secret, since the ignition-server/tokensecret_controller does not control it
+			if hcAnnotation, exists := typedObj.Annotations[hyperutil.HostedClusterAnnotation]; exists {
+				if nodePoolAnnotation, exists := typedObj.Annotations[hyperv1.NodePoolLabel]; exists {
```

Comment:
```suggestion
				if nodePoolAnnotation, exists := typedObj.Annotations[hyperv1.NodePoolِAnnotation]; exists {
```
can we please use the new constant here as well